src/Fixer/PhpUnit/PhpUnitDataProviderReturnTypeFixer.php

```diff
@@ -0,0 +1,119 @@
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ */
+final class PhpUnitDataProviderReturnTypeFixer extends AbstractPhpUnitFixer
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'The return type of PHPUnit data provider must be `iterable`.',
+            [
+                new CodeSample(
+                    '<?php
+class FooTest extends TestCase {
+    /**
+     * @dataProvider provideSomethingCases
+     */
+    public function testSomething($expected, $actual) {}
+    public function provideSomethingCases(): array {}
+}
+',
+                ),
+                new CodeSample(
+                    '<?php
+class FooTest extends TestCase {
+    /**
+     * @dataProvider provideSomethingCases
+     */
+    public function testSomething($expected, $actual) {}
+    public function provideSomethingCases() {}
+}
+',
+                ),
+            ],
+            'Data provider must return `iterable`, either an array of arrays or an object that implements the `Traversable` interface.',
+            'Risky when relying on signature of the data provider.',
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before ReturnTypeDeclarationFixer.
+     * Must run after CleanNamespaceFixer.
+     */
+    public function getPriority(): int
+    {
+        return 0;
+    }
+
+    public function isRisky(): bool
+    {
+        return true;
+    }
+
+    protected function applyPhpUnitClassFix(Tokens $tokens, int $startIndex, int $endIndex): void
+    {
+        $dataProviderAnalyzer = new DataProviderAnalyzer();
+        $functionsAnalyzer = new FunctionsAnalyzer();
+
+        foreach (array_reverse($dataProviderAnalyzer->getDataProviders($tokens, $startIndex, $endIndex)) as $dataProviderAnalysis) {
+            $typeAnalysis = $functionsAnalyzer->getFunctionReturnType($tokens, $dataProviderAnalysis->getNameIndex());
+
+            if (null === $typeAnalysis) {
+                $argumentsStart = $tokens->getNextTokenOfKind($dataProviderAnalysis->getNameIndex(), ['(']);
+                $argumentsEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $argumentsStart);
+
+                $tokens->insertAt(
+                    $argumentsEnd + 1,
+                    [
+                        new Token([CT::T_TYPE_COLON, ':']),
+                        new Token([T_WHITESPACE, ' ']),
+                        new Token([T_STRING, 'iterable']),
+                    ],
+                );
+
+                continue;
+            }
+
+            if ('iterable' === $typeAnalysis->getName()) {
+                continue;
+            }
+
+            $typeStartIndex = $tokens->getNextMeaningfulToken($typeAnalysis->getStartIndex() - 1);
+            $typeEndIndex = $typeAnalysis->getEndIndex();
+
+            if ($tokens->generatePartialCode($typeStartIndex, $typeEndIndex) !== $typeAnalysis->getName()) {
+                continue;
```

Comment:
```suggestion
                // continue;
```

test - is it only for 7.4?